src/plugins/codeStyle/index.ts

```diff
@@ -164,6 +170,24 @@ export default class CodeStyle {
         file.addDiagnostics(bsDiagnostics);
     }
 
+    validateColorFormat(fileContents: string, diagnostics: (Omit<BsDiagnostic, 'file'>)[], colorFormat: RuleColorFormat) {
```

Comment:
You shouldn't be operating on the `fileContents`. We've parsed the entire file into AST, so that's how bslint should be navigating all the various tokens, expressions, and statements. You can tap into the `.walk()` visitor function above and add callouts for whatever expressions you're interested in. Things like `LiteralExpression` hold raw string values for brs/bs files.